vendor/bat-native-ledger/src/bat/ledger/internal/database/database_publisher_prefix_list.cc

```diff
@@ -0,0 +1,204 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/database/database_publisher_prefix_list.h"
+
+#include <tuple>
+#include <utility>
+
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/stringprintf.h"
+#include "bat/ledger/internal/database/database_util.h"
+#include "bat/ledger/internal/publisher/prefix_util.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/state/state_keys.h"
+
+using std::placeholders::_1;
+using braveledger_publisher::PrefixIterator;
+
+namespace {
+
+const char kTableName[] = "publisher_prefix_list";
+
+constexpr size_t kHashPrefixSize = 4;
+constexpr size_t kMaxInsertRecords = 100'000;
+
+std::tuple<PrefixIterator, std::string, size_t> GetPrefixInsertList(
```

Comment:
can you please add unit tests for this one

Reply:
Since this is internal, I chose to add a unit test for the "Reset" method, which tests how large inserts are split up between transactions.

Reply:
Sounds good